student-projects/gregory/Sdutents-Stats/app.ts

```diff
@@ -0,0 +1,59 @@
+let StatType = 0;
+let Answer = "";
+while (!((StatType === 1) || (StatType === 2)))  {
+    Answer = prompt("Student Statistics. Do you want '1' - Best Grade or '2' - Worst Grade?");
+    if (Answer === null) {
+        alert("Looks like you don't want any grade. Good Day!");
+        throw "Canceled by the user.";
+    }
+    StatType = Number(Answer);
+    if (!((StatType === 1) || (StatType === 2))){
+        alert("ERROR - wrong input, it must be 1 or 2!");    
+    } 
+}
+
+let MinGrade = 100;
+let MaxGrade = 0;
+let StudentName = "";
+let NameAnswer = "";
+while (Answer !== null){
+    Answer = prompt("Enter Student's Full Name. Cancel to stop.");
+    if (Answer === null) {
+        break;
+    }
+    
+    NameAnswer = Answer;
+       
+    let CurrGrade = -1;
+    while ((CurrGrade < 0) || (CurrGrade > 100)){
```

Comment:
This condition is not sufficient, in line 34 we set `CurrGrade` to what is possibly `NaN`.
In this case, both conditions would evaluate to `false` and we'll exit the loop.